packages/http/src/decorators.ts

```diff
@@ -663,3 +664,54 @@ export function includeInapplicableMetadataInPayload(
   }
   return true;
 }
+
+const requestVisibilityKey = createStateSymbol("requestVisibility");
+
+export function $requestVisibility(
+  context: DecoratorContext,
+  target: Operation,
+  ...visibilities: string[]
+) {
+  validateDecoratorUniqueOnNode(context, target, $requestVisibility);
+  const state = context.program.stateMap(requestVisibilityKey);
+  state.set(target, visibilities);
+}
+
+/**
+ * Returns the visibility of the given operation, if provided with `@requestVisibility`.
+ *
+ * @see $requestVisibility
+ */
+export function getOperationRequestVisibility(
```

Comment:
Couldn't reuse the name `getRequestVisibility` because it conflicts with the one with verb as a param.